plugin.video.last_played/default.py

```diff
@@ -0,0 +1,206 @@
+		os.makedirs(txtpath)
+txtfile = txtpath + "lastPlayed.json"
+fivestar = addon.getSetting('fivestar')
+enable_debug = addon.getSetting('enable_debug')
+lang = addon.getLocalizedString
+
+# Builds JSON request with provided json data
+def buildRequest(method, params, jsonrpc='2.0', rid='1'):
+	request = { 'jsonrpc' : jsonrpc, 'method' : method, 'params' : params, 'id' : rid }
+	return request
+
+# Checks JSON response and returns boolean result
+def checkReponse(response):
+	result = False
+	if ( ('result' in response) and ('error' not in response) ):
+		result = True
+	return result
+
+# Executes JSON request and returns the JSON response
+def JSexecute(request):
+	request_string = json.dumps(request)
+	response = xbmc.executeJSONRPC(request_string)
+	if ( response ):
+		response = json.loads(response)
+	return response
+
+# Performs single JSON query and returns result boolean, data dictionary and error string
+def JSquery(request):
+	result = False
+	data = {}
+	error = ''
+	if ( request ):
+		response = JSexecute(request)
+		if ( response ):
+			result = checkReponse(response)
+			if ( result ):
+				data = response['result']
+			else: error = response['error']
+	return (result, data, error)
+
+def send2fivestar(line):
+	wid = int(line["id"])
+	if line["type"]=="movie": typ="M"
+	elif line["type"]=="episode": typ="S"
+	else: typ="V"
+
+	if typ=="M" and addon.getSetting('movies') != "true": return
+	if typ=="S" and addon.getSetting('tv') != "true": return
+	if typ=="V" and addon.getSetting('videos') != "true": return
+
+	imdbId = ""
+	tvdbId = ""
+	orgTitle = ""
+	showTitle = line["show"]
+	season = line["season"]
+	episode = line["episode"]
+	thumbnail = line["thumbnail"]
+	fanart = line["fanart"]
+	if wid>0:
+		if typ=="M":
+			request = buildRequest('VideoLibrary.GetMovieDetails', {'movieid' : wid, 'properties' : ['imdbnumber', 'originaltitle']})
+			result, data = JSquery(request)[:2]
+			if ( result and 'moviedetails' in data ):
+				imdbId = data['moviedetails']["imdbnumber"]
+				orgTitle = data['moviedetails']["originaltitle"]
+		elif typ=="S":
+			request = buildRequest('VideoLibrary.GetEpisodeDetails', {'episodeid' : wid, 'properties' : ['tvshowid', 'season', 'episode']})
+			result, data = JSquery(request)[:2]
+			if ( result and 'episodedetails' in data ):
+				season = data['episodedetails']["season"]
+				episode = data['episodedetails']["episode"]
+				request = buildRequest('VideoLibrary.GetTvShowDetails', {'tvshowid' : data['episodedetails']["tvshowid"], 'properties' : ['imdbnumber', 'originaltitle']})
+				result, data = JSquery(request)[:2]
+				if ( result and 'tvshowdetails' in data ):
+					showTitle = data['tvshowdetails']["label"]
+					orgTitle = data['tvshowdetails']["originaltitle"]
+					tvdbId = data['tvshowdetails']["imdbnumber"]
+
+	xvideo = line["file"]
+	if "video" in line and line["video"]!="": xvideo = line["video"]
+	url = "http://www.5star-movies.com/WebService.asmx/kodiWatch?tmdbId="
```

Comment:
It's a site that I am making. It will be a frontend for tmdb, but also can keep track of the watched history.

Optionally on the addon, we can choose to activate it, and then see the watched history on the site. By default if off.